### DISCUSSION TITLE
Referencing images

### DISCUSSION THREAD

**Discussion by jennyberger-writer** [skoropadas/ng-doc#161] [Q&A] [answered]
Hello -- I'm a tech writer on a team that's using NgDoc for both internal dev docs & customer-facing product docs. So, I'm rather new to Angular and NgDocs, but I'm learning. :-) 

I'm having trouble referencing images within a page. I've tried to use standard Markdown & HTML to reference an image, but when I preview locally, the image doesn't render. So, I'd greatly appreciate some help with the following questions:

- Do I need to create a page processor for images, as described here: https://ng-doc.com/docs/features/page-processors
- If so, does the code for registering the page processor change in some way if I want to enable it for all pages, by adding it to main.ts? In the Page Processors article, the code given appears to be only for registering the processor in a specific page.
- I'm assuming my image directory should be a child of ng-doc/ProjectName/assets (per the angular.json file). Is this correct? 

Thanks for your help!!

**Comment by skoropadas** [comment, marked as answer]
Hi Jenny, I'm glad your team chose NgDoc.

Regarding your question:
- PageProcessor is not required for displaying images. It is intended for cases when you want to customize or add interactivity to default HTML elements.
- The NgDoc assets included in your angular.json connect icons and additional static files. They do not affect overall performance or your usage, but you should ensure that the `assets` folder of your application is also listed in the assets in `angular.json` (e.g., `src/assets`).

To display an image you should simply specify the path to the static files of your application, for example:
  - Place the required images in the `src/assets` directory or one of its subdirectories (`src/assets/my-image.png`)
  - Inside the documentation, you should use the path relative to the `src` directory, like this - `![alt text](assets/my-image.png 'Image title')`
  
 Sometimes after adding new files to the assets, Angular does not see the changes and does not update them within the dev server. Therefore, if you have added new images but they are not displaying, it is best to try restarting the dev server.

**Comment by jennyberger-writer** [comment]
Thank you so much! I really appreciate your kind attention and taking the time to help. :-)
